src/shared/sam/activation.ts

```diff
@@ -249,12 +250,16 @@ async function activateCodeLensProviders(
  * Will show once per extension activation at most (all prompting triggers are disposed of on first trigger)
  * Will not show if the YAML extension is installed or if a user has permanently dismissed the message.
  */
-function createYamlExtensionPrompt(): void {
-    const neverPromptAgain = ext.context.globalState.get<boolean>(STATE_NAME_SUPPRESS_YAML_PROMPT)
+async function createYamlExtensionPrompt(): Promise<void> {
```

Comment:
The call site was not updated to `await` this. This function checks `vscode.window.visibleTextEditors` so there might be a race. When changing functions to `async`, always check their call sites.